pkg/controllers/cspi-controller/handler.go

```diff
@@ -67,6 +67,16 @@ func (c *CStorPoolInstanceController) reconcile(key string) error {
 	}
 	cspi = cspiObj
 
+	// validate CSPI validates the CSPI
+	err = validateCSPI(cspi)
+	if err != nil {
+		c.recorder.Event(cspi,
+			corev1.EventTypeWarning,
+			"Validation failed",
```

Comment:
```
"Validation failed for cspi %s"
```
The err.Error() should only return the specific validation errors and the functions underneath not worry about putting the cspi name.

Reply:
Basically event will log the cspi name right still we need to log it? I can do that in the message instead of `Reason` is that make sense?